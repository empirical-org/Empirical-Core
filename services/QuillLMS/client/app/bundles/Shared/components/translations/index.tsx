import LanguagePicker from './languagePicker'
import LanguageSelectionPage from './languageSelectionPage'

export { LanguagePicker, LanguageSelectionPage }
