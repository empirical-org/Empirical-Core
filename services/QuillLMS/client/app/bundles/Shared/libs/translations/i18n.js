import i18n from "i18next";
import { initReactI18next } from "react-i18next";

import translationAR from './modules/arabic/translation.json';
import translationCN from './modules/chinese/translation.json';
import translationDR from './modules/dari/translation.json';
import translationEN from './modules/english/translation.json';
import translationFR from './modules/french/translation.json';
import translationDE from './modules/german/translation.json';
import translationHI from './modules/hindi/translation.json';
import translationJP from './modules/japanese/translation.json';
import translationKR from './modules/korean/translation.json';
import translationPT from './modules/portuguese/translation.json';
import translationRU from './modules/russian/translation.json';
import translationES from './modules/spanish/translation.json';
import translationTG from './modules/tagalog/translation.json';
import translationTH from './modules/thai/translation.json';
import translationUK from './modules/ukrainian/translation.json';
import translationUR from './modules/urdu/translation.json';
import translationVT from './modules/vietnamese/translation.json';

// the translations
const resources = {
  arabic: { translation: translationAR },
  chinese: { translation: translationCN },
  dari: { translation: translationDR },
  english: { translation: translationEN },
  french: { translation: translationFR },
  german: { translation: translationDE },
  hindi: { translation: translationHI },
  japanese: { translation: translationJP},
  korean: { translation: translationKR},
  portuguese: { translation: translationPT},
  russian: { translation: translationRU},
  spanish: { translation: translationES },
  tagalog: { translation: translationTG },
  thai: { translation: translationTH },
  ukrainian: { translation: translationUK },
  urdu: { translation: translationUR },
  vietnamese: { translation: translationVT }
};

//i18n configurations

i18n
  .use(initReactI18next)
  .init({
    resources,
    lng: "english",
    defaultTransParent: 'div',
    // we use the ^ and | as separators because many label strings contain a period or colon,
    // which will cause the translation to fail
    keySeparator: '^',
    nsSeparator: '|',
    interpolation: {
      escapeValue: false
    }
  });

export default i18n;
