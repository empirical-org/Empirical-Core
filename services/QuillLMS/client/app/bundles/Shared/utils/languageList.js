import languageConstants from '../../../../../config/locales/languageConstants.json';

export const {
  ENGLISH,
  CHINESE,
  HINDI,
  SPANISH,
  FRENCH,
  ARABIC,
  RUSSIAN,
  PORTUGUESE,
  URDU,
  GERMAN,
  JAPANESE,
  KOREAN,
  VIETNAMESE,
  THAI,
  UKRAINIAN,
  TAGALOG,
  DARI
} = languageConstants;

export const languages = [ENGLISH, CHINESE, HINDI, SPANISH, FRENCH, ARABIC, RUSSIAN, PORTUGUESE, URDU, GERMAN, JAPANESE, KOREAN, VIETNAMESE, THAI, UKRAINIAN, TAGALOG, DARI];
export const rightToLeftLanguages = [ARABIC, URDU, DARI];

export const diagnosticLanguageOptions = {
  [ENGLISH]: {
    flag: 'https://assets.quill.org/images/flags/usa.png',
    label: 'English'
  },
  [CHINESE]: {
    flag: 'https://assets.quill.org/images/flags/china.png',
    label: '中文'
  },
  [HINDI]: {
    flag: 'https://assets.quill.org/images/flags/india.png',
    label: 'हिंदी'
  },
  [SPANISH]: {
    flag: 'https://assets.quill.org/images/flags/spain.png',
    label: 'Español'
  },
  [FRENCH]: {
    flag: 'https://assets.quill.org/images/flags/france.png',
    label: 'Français'
  },
  [ARABIC]: {
    flag: 'https://assets.quill.org/images/flags/egypt.png',
    label: 'العربية'

  },
  [RUSSIAN]: {
    flag: 'https://assets.quill.org/images/flags/russia.png',
    label: 'Русский'

  },
  [PORTUGUESE]: {
    flag: 'https://assets.quill.org/images/flags/brazil.png',
    label: 'Português'

  },
  [URDU]: {
    flag: 'https://assets.quill.org/images/flags/pakistan.png',
    label: 'اردو'
  },
  [GERMAN]: {
    flag: 'https://assets.quill.org/images/flags/germany.png',
    label: 'Deutsch'
  },
  [JAPANESE]: {
    flag: 'https://assets.quill.org/images/flags/japan.png',
    label: '日本語'
  },
  [KOREAN]: {
    flag: 'https://assets.quill.org/images/flags/south_korea.png',
    label: '한국어'
  },
  [VIETNAMESE]: {
    flag: 'https://assets.quill.org/images/flags/vietnam.png',
    label: 'Tiếng Việt'
  },
  [THAI]: {
    flag: 'https://assets.quill.org/images/flags/thailand.png',
    label: 'ไทย'
  },
  [UKRAINIAN]: {
    flag: 'https://assets.quill.org/images/flags/ukraine.png',
    label: 'Українська'
  },
  [TAGALOG]: {
    flag: 'https://assets.quill.org/images/flags/philippines.png',
    label: 'Tagalog'
  },
  [DARI]: {
    flag: 'https://assets.quill.org/images/flags/afghanistan.png',
    label: 'درى'
  },
}
