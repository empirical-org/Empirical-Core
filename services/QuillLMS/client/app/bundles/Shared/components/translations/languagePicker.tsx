import * as React from 'react';

import { DropdownInput } from '../../index';
import { defaultLanguageOptions, defaultLanguages } from "../../utils/languageList";

interface LanguagePickerProps {
  language: string,
  languageOptions: any,
  updateLanguage(language: string): any
}

const LanguagePicker = ({ language, updateLanguage, languageOptions }: LanguagePickerProps) => {
  const options = (): Array<{value: string, label: string}> =>
    languageOptions ?? defaultLanguages.map(language => ({
      value: language,
      label: defaultLanguageOptions[language].label
    }));

  const onChange = (option: { value: string}) => {
    const language = option.value;
    updateLanguage(language)
  }

  const value = options().find(opt => language === opt.value)

  return (
    <DropdownInput
      className="ell-language-selector medium borderless"
      handleChange={onChange}
      options={options()}
      value={value}
    />
  )
}

export default LanguagePicker;
