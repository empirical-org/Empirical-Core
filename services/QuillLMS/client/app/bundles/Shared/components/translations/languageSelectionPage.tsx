import React from 'react';

import { defaultLanguageOptions, ENGLISH } from '../../utils/languageList';
import { TrackAnalyticsEvent } from '../../../Diagnostic/actions/analytics';
import { Events } from '../../../Diagnostic/modules/analytics';

interface LanguageSelectionPageProps {
  questionCount?: number,
  dispatch: (any) => void,
  setLanguage: (language: string) => void,
  previewMode: boolean,
  begin: () => void,
  languages: string[]
}

export const LanguageSelectionPage = ({
  questionCount,
  dispatch,
  setLanguage,
  previewMode,
  begin,
  languages
}: LanguageSelectionPageProps) => {

  function handleClickLanguage (e) {
    const language = e.currentTarget.value;
    const isDiagnosticActivity = window.location.href.includes('diagnostic')
    if (isDiagnosticActivity && language !== ENGLISH) {
      dispatch(TrackAnalyticsEvent(Events.DIAGNOSTIC_LANGUAGE_SELECTED, { language }));
    }
    setLanguage(language);
    if (previewMode) {
      begin();
    }
  }
  return (
    <div className="language-page">
      <div className="introductory-text">
        {questionCount && <p>{`Hello there! You are about to start a ${questionCount} question placement activity.`}</p>}
        <p>First, let's set up your language preference. All the directions are in English by default.</p>
        <p>Show directions in English only.</p>
      </div>
      <div className="language-button-container english">
        <button className="language-button" onClick={handleClickLanguage} type="button" value="english">
          <img alt="flag" className="language-button-img" src={defaultLanguageOptions[ENGLISH].flag} />
          <p className="language-label">English</p>
        </button>
      </div>
      <div className="divider-container">
        <div className="divider" />
        <p className="divider-label">or</p>
        <div className="divider" />
      </div>
      <div className="introductory-text">
        <p>Show directions in English <span>and</span> another language.</p>
      </div>
      <div className="language-button-container">
        {languages.map(language => {
          if (language !== ENGLISH) {
            return (
              <button className="language-button" key={`${language}-button`} onClick={handleClickLanguage} type="button" value={language}>
                <img alt="flag" className="language-button-img" src={defaultLanguageOptions[language].flag} />
                <p className="language-label">{defaultLanguageOptions[language].label}</p>
              </button>
            );
          }
        })}
      </div>
    </div>
  );
}

export default LanguageSelectionPage;
