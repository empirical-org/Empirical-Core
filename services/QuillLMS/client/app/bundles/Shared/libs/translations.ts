import { ENGLISH } from '../utils/languageList'

export const getlanguageOptions = (translations) => ([
  { value: ENGLISH, label: ENGLISH },
  ...Object.keys(translations).map(language => ({
    value: language,
    label: language
  }))
]);
