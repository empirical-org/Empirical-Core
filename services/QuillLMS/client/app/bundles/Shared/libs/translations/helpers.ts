import { ENGLISH } from '../../utils/languageList'

export const getlanguageOptions = (translations) => ([
  { value: ENGLISH, label: ENGLISH },
  ...Object.keys(translations).map(language => ({
    value: language,
    label: language
  }))
]);

// Temporary feature flag until we are ready to ship this.
export const hasTranslationFlag = (): boolean => {
  const urlParams = new URLSearchParams(window.location.search)
  if (urlParams.get('showTranslations') !== 'true') return false;
  return true;
};

export const showTranslations = (language, languageOptions): boolean => {
  return hasTranslationFlag() && !!languageOptions && !!language && Object.keys(languageOptions).length > 1;
};
