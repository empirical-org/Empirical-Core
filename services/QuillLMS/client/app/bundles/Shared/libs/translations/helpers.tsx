import * as React from 'react';
import { ENGLISH } from '../../utils/languageList'
import { defaultLanguages } from '../..';

export const getlanguageOptions = (translations) => ([
  { value: ENGLISH, label: ENGLISH },
  ...Object.keys(translations).map(language => ({
    value: language,
    label: language
  }))
]);

interface renderSaveAndExitArguments {
  isELLDiagnostic: boolean,
  language: string,
  languageOptions?: { label: string, value: string }[],
  translate: (language: string) => string
}

export function renderSaveAndExitButton({ isELLDiagnostic, language, languageOptions, translate }: renderSaveAndExitArguments) {
  const languages = isELLDiagnostic ? defaultLanguages : languageOptions && languageOptions.map(option => option.value)
  let buttonText = 'Save and exit'
  if (language && language !== ENGLISH && languages && languages.includes(language)) {
    buttonText = translate('buttons^save and exit')
  }
  return <a className="quill-button medium contained white focus-on-dark" href={`${process.env.DEFAULT_URL}/profile`}>{buttonText}</a>
}

// Temporary feature flag until we are ready to ship this.
export const hasTranslationFlag = (): boolean => {
  const urlParams = new URLSearchParams(window.location.search)
  if (urlParams.get('showTranslations') !== 'true') return false;
  return true;
};

export const showTranslations = (language, languageOptions): boolean => {
  return hasTranslationFlag() && !!languageOptions && !!language && Object.keys(languageOptions).length > 1;
};
