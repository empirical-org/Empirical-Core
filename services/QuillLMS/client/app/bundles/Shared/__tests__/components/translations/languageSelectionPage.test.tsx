import * as React from "react";
import { render } from "@testing-library/react";
import { defaultLanguages } from '../../../utils/languageList'

import { LanguageSelectionPage } from "../../../components/translations/languageSelectionPage";

const mockProps = {
  dispatch: jest.fn(),
  setLanguage: jest.fn(),
  previewMode: false,
  begin: jest.fn(),
  languages: defaultLanguages
}

describe('LanguageSelectionPage', () => {
  test('it should render', () => {
    const { asFragment } = render(<LanguageSelectionPage {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  })
})
