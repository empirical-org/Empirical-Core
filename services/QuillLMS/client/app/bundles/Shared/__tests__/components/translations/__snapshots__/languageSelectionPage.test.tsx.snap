// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelectionPage it should render 1`] = `
<DocumentFragment>
  <div
    class="language-page"
  >
    <div
      class="introductory-text"
    >
      <p>
        First, let's set up your language preference. All the directions are in English by default.
      </p>
      <p>
        Show directions in English only.
      </p>
    </div>
    <div
      class="language-button-container english"
    >
      <button
        class="language-button"
        type="button"
        value="english"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/usa.png"
        />
        <p
          class="language-label"
        >
          English
        </p>
      </button>
    </div>
    <div
      class="divider-container"
    >
      <div
        class="divider"
      />
      <p
        class="divider-label"
      >
        or
      </p>
      <div
        class="divider"
      />
    </div>
    <div
      class="introductory-text"
    >
      <p>
        Show directions in English 
        <span>
          and
        </span>
         another language.
      </p>
    </div>
    <div
      class="language-button-container"
    >
      <button
        class="language-button"
        type="button"
        value="chinese"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/china.png"
        />
        <p
          class="language-label"
        >
          中文
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="hindi"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/india.png"
        />
        <p
          class="language-label"
        >
          हिंदी
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="spanish"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/spain.png"
        />
        <p
          class="language-label"
        >
          Español
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="french"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/france.png"
        />
        <p
          class="language-label"
        >
          Français
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="arabic"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/egypt.png"
        />
        <p
          class="language-label"
        >
          العربية
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="russian"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/russia.png"
        />
        <p
          class="language-label"
        >
          Русский
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="portuguese"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/brazil.png"
        />
        <p
          class="language-label"
        >
          Português
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="urdu"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/pakistan.png"
        />
        <p
          class="language-label"
        >
          اردو
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="german"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/germany.png"
        />
        <p
          class="language-label"
        >
          Deutsch
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="japanese"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/japan.png"
        />
        <p
          class="language-label"
        >
          日本語
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="korean"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/south_korea.png"
        />
        <p
          class="language-label"
        >
          한국어
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="vietnamese"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/vietnam.png"
        />
        <p
          class="language-label"
        >
          Tiếng Việt
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="thai"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/thailand.png"
        />
        <p
          class="language-label"
        >
          ไทย
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="ukrainian"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/ukraine.png"
        />
        <p
          class="language-label"
        >
          Українська
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="tagalog"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/philippines.png"
        />
        <p
          class="language-label"
        >
          Tagalog
        </p>
      </button>
      <button
        class="language-button"
        type="button"
        value="dari"
      >
        <img
          alt="flag"
          class="language-button-img"
          src="https://assets.quill.org/images/flags/afghanistan.png"
        />
        <p
          class="language-label"
        >
          درى
        </p>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
